opal/mca/patcher/base/patcher_base_patch.c

```diff
@@ -138,10 +141,24 @@ static inline void apply_patch (unsigned char *patch_data, uintptr_t address, si
 {
     ModifyMemoryProtection (address, data_size, PROT_EXEC|PROT_READ|PROT_WRITE);
     memcpy ((void *) address, patch_data, data_size);
-    for (size_t i = 0 ; i < data_size ; i += 16) {
+#if HAVE___CLEAR_CACHE
+    /* do not allow global declaration of compiler intrinsic */
+    void __clear_cache(void* beg, void* end);
+
+    __clear_cache ((void *) address, (void *) (address + data_size));
+#else
+    size_t offset_jump = 16;
+
+#if OPAL_ASSEMBLY_ARCH == OPAL_ARM64
+    offset_jump = 4;
```

Comment:
how is the offsets are calculated ? (4, 16)

Reply:
Double checking ...

Reply:
Arm (Section D3.4) advices to use cache line increments obtained from ```ctr_el0``` (DminLine, IminLine fields),similar to: https://github.com/openucx/ucx/blob/bac22184987518b990b006bf9f02bbff46745d28/src/ucs/arch/aarch64/cpu.h#L144 

it is log2 based value. I think the minimum value for Armv8 is 32 bytes (16 words) but your setup most likely to have 64 bytes. I guess, 32 will work but you just hitting the same d/i-cacheline twice.

Reply:
Ok, I see. I will just use 32-bytes then. Not like this routine is being run too often :).

Reply:
Ok, pushed.